File: Csharp/Leetcode/Utils/TreeNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
namespace Leetcode.Utils {
    public class TreeNode {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode() {}
        public TreeNode(int x) { val = x; }

        public override string ToString() {
            var sList = new List<string>();
            var queue = new Queue<TreeNode>();
            queue.Enqueue(this);

            var nullCount = 0;
            while (queue.Count > 0) {
                var node = queue.Dequeue();

                if (node == null) {
                    nullCount++;
                } else {
                    if (nullCount > 0) {
                        for (int i = 0; i < nullCount; i++) {
                            sList.Add("null");
                        }
                        nullCount = 0;
                    }            
                    sList.Add(node.val.ToString());
                    queue.Enqueue(node.left);
                    queue.Enqueue(node.right);
                }
            }

            return sList.Aggregate((i, j) => i + ", " + j);
        }
    }
}

File: Java/Leetcode/src/main/java/com/leetcode/utils/TreeNode.java
package com.leetcode.utils;

import java.util.*;

public class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode(int x) { val = x; }

    @Override
    public String toString() {
        List<String> sList = new ArrayList<>();
        Deque<TreeNode> queue = new LinkedList<>();
        queue.add(this);

        int nullCount = 0;
        while (queue.size() > 0) {
            TreeNode node = queue.remove();

            if (node == null) {
                nullCount++;
            } else {
                if (nullCount > 0) {
                    for (int i = 0; i < nullCount; i++) {
                        sList.add("null");
                    }
                    nullCount = 0;
                }
                sList.add(String.valueOf(node.val));
                queue.add(node.left);
                queue.add(node.right);
            }
        }

        return String.join(", ", sList);
    }
}
